frontend/libs/studio-components/src/components/StudioTableLocalPagination/StudioTableLocalPagination.tsx

```diff
@@ -1,75 +1,64 @@
-    itemLabel: (num: number) => string;
-  };
+  emptyTableFallback?: React.ReactNode;
+  pagination?: LocalPaginationProps;
 };
 
 export const StudioTableLocalPagination = forwardRef<
   HTMLTableElement,
   StudioTableLocalPaginationProps
->(
-  (
-    { columns, rows, isSortable = true, size = 'medium', emptyTableMessage, pagination },
-    ref,
-  ): React.ReactElement => {
-    const [currentPage, setCurrentPage] = useState<number>(1);
-    const [pageSize, setPageSize] = useState<number>(pagination?.pageSizeOptions[0] ?? undefined);
+>(({ columns, rows, size = 'medium', emptyTableFallback, pagination }, ref): React.ReactElement => {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [pageSize, setPageSize] = useState<number>(pagination?.pageSizeOptions[0] ?? undefined);
 
-    const { handleSorting, sortedRows } = useTableSorting(rows, { enable: isSortable });
+  const isSortable = columns.some((column) => column.sortable);
+  const { handleSorting, sortedRows } = useTableSorting(rows, { enable: isSortable });
 
-    const initialRowsToRender = getRowsToRender(currentPage, pageSize, rows);
-    const [rowsToRender, setRowsToRender] = useState<Rows>(initialRowsToRender);
+  const initialRowsToRender = getRowsToRender(currentPage, pageSize, rows);
+  const [rowsToRender, setRowsToRender] = useState<Rows>(initialRowsToRender);
 
-    useEffect(() => {
-      const newRowsToRender = getRowsToRender(currentPage, pageSize, sortedRows || rows);
+  useEffect(() => {
+    const newRowsToRender = getRowsToRender(currentPage, pageSize, sortedRows || rows);
+    setRowsToRender(newRowsToRender);
+  }, [sortedRows, rows, currentPage, pageSize]);
 
-      const isOutOfRange = !newRowsToRender.length && currentPage > 1;
-      if (isOutOfRange) {
-        setCurrentPage(1);
-        setRowsToRender(getRowsToRender(1, pageSize, sortedRows || rows));
-        return;
-      }
+  const totalRows = rows.length;
+  const totalPages = Math.ceil(totalRows / pageSize);
 
-      setRowsToRender(newRowsToRender);
-    }, [sortedRows, rows, currentPage, pageSize]);
-
-    const totalPages = Math.ceil(rows.length / pageSize);
-
-    const studioTableRemotePaginationProps = pagination && {
-      ...pagination,
-      currentPage,
-      totalPages,
-      onPageChange: setCurrentPage,
-      onPageSizeChange: setPageSize,
-    };
+  const studioTableRemotePaginationProps = pagination && {
+    ...pagination,
+    pageSize,
+    currentPage,
+    totalPages,
+    totalRows,
+    onPageChange: setCurrentPage,
+    onPageSizeChange: setPageSize,
+  };
 
-    return (
-      <StudioTableRemotePagination
-        columns={columns}
-        rows={rowsToRender}
-        size={size}
-        emptyTableMessage={emptyTableMessage}
-        onSortClick={isSortable && handleSorting}
-        pagination={studioTableRemotePaginationProps}
-        ref={ref}
-      />
-    );
-  },
-);
+  return (
+    <StudioTableRemotePagination
```

Comment:
StudioTableRemote and StudioTablelocal seems very similar. Could they be the same component where the prop input are differing? looks like its the sorting mechanism that differs only 🤔

Reply:
If we did, we would get quite a big component, with many props and conditionals to take care of... I like the nested approach that we have now better